src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedCatchParameterShouldBeUnnamedCheck.java

```diff
@@ -0,0 +1,307 @@
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_CATCH,
+            TokenTypes.IDENT,
+        };
+    }
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        catchParameter = null;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ast.getType() == TokenTypes.LITERAL_CATCH) {
+            visitCatchParameter(ast);
+        }
+        else {
+            visitIdent(ast);
+        }
+    }
+
+    @Override
+    public void leaveToken(DetailAST ast) {
+        logUnusedCatchParameter(ast);
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#LITERAL_CATCH}
+     * and set the catch parameter.
+     *
+     * @param catchAst token representing {@link TokenTypes#LITERAL_CATCH}
+     */
+    private void visitCatchParameter(DetailAST catchAst) {
+        final DetailAST catchParameterAst = catchAst.findFirstToken(TokenTypes.PARAMETER_DEF);
+        final String catchParameterName =
+                catchParameterAst.findFirstToken(TokenTypes.IDENT).getText();
+
+        if (!"_".equals(catchParameterName)) {
+            catchParameter = new CatchParameter(catchParameterName,
+                                    catchAst, catchParameterAst);
+        }
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#IDENT}
+     * and check if it is a candidate for a catch parameter usage.
+     *
+     * @param identAST token representing {@link TokenTypes#IDENT}
+     */
+    private void visitIdent(DetailAST identAST) {
+        final boolean isCatchParameter = identAST.getParent().getParent()
+                                                .getType() == TokenTypes.LITERAL_CATCH;
+
+        if (!isCatchParameter
+                && (!TokenUtil.isOfType(identAST.getParent(), UNACCEPTABLE_PARENT_OF_IDENT)
+                        || shouldCheckDotChild(identAST))) {
+            checkIdentifier(identAST);
+        }
+    }
+
+    /**
+     * Check if the given {@link TokenTypes#IDENT} is a child of a dot operator
+     * and should be checked for usage.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     * @return true if the given {@link TokenTypes#IDENT} is a child of a dot operator
+     *     and should be checked for usage.
+     */
+    private static boolean shouldCheckDotChild(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return parent.getType() == TokenTypes.DOT
+                && identAst == parent.getFirstChild();
+    }
+
+    /**
+     * Check the given {@link TokenTypes#IDENT} and
+     * register it as used if it is the catch parameter.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     */
+    private void checkIdentifier(DetailAST identAst) {
+        if (catchParameter != null
+                && !isLeftHandSideValue(identAst)
```

Comment:
So we don't count reassignment as usage? Judging by the test cases below, we currently do. What am I missing?

Reply:
Yes we don't count reassignment as usage. here is its test case : https://github.com/checkstyle/checkstyle/blob/c50d03687247b8f5e303845ded14a7ef0d346abe/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unusedcatchparametershouldbeunnamed/InputUnusedCatchParameterShouldBeUnnamed.java#L49-L53


>  Judging by the test cases below, we currently do.

can you point to this?

Reply:
Note I don't consider the test case below to be a reassignment.

https://github.com/checkstyle/checkstyle/blob/c50d03687247b8f5e303845ded14a7ef0d346abe/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unusedcatchparametershouldbeunnamed/InputUnusedCatchParameterShouldBeUnnamed.java#L55-L60

Reply:
https://github.com/checkstyle/checkstyle/pull/15358#discussion_r1691830997 is weird to me.If we were to make `e` unnamed, we would have to remove this reassignment (even though it does nothing). I suppose I am good with this, let's get this check into the hands of users and go from there.

Reply:
Leaving this one unresolved so others can see.